docs/transforms.rst

```diff
@@ -0,0 +1,20 @@
+
+.. _transforms-caveats:
+
+Limitations of using JAX transforms inside of networks
+======================================================
+
+Once a Haiku network has been transformed to a pure function using
+:func:`hk.transform() <haiku.transform>`, it's possible to freely combine it with
+any JAX transformations like :func:`jax.jit`, :func:`jax.grad`, and so on.
+It's also possible to use JAX transformations inside of a Haiku network, but
+there can be issues with JAX's tracing mechanism since :func:`hk.transform()
+<haiku.transform>` has side effects.
+To work around this, Haiku provides wrapped versions of JAX transforms under
+the :mod:`haiku` namespace. You can access these as :func:`hk.jit() <haiku.jit>`,
+:func:`hk.grad() <haiku.grad>`, and so on.
+
+These wrappers work by calling :func:`hk.transform() <haiku.transform>` on the
```

Comment:
This isn't actually how they work, unfortunately. Consider something like:
"These wrappers functionalize the underlying function by causing them to take Haiku state as input and return the (potentially modified) Haiku as output. They then apply the corresponding JAX transform...."

Reply:
I changed it to "turns the underlying Haiku function into a pure function", which I think is easier to understand and sufficiently vague to be correct.

Reply:
Sounds reasonable, change isn't manifested.